feature-libs/order/components/order-details/order-detail-reorder/reorder-dialog/reorder-dialog.component.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import {
+  CartModification,
+  CartValidationStatusCode,
+} from '@spartacus/cart/base/root';
+import {
+  FocusConfig,
+  ICON_TYPE,
+  LaunchDialogService,
+} from '@spartacus/storefront';
+import { BehaviorSubject, Subscription } from 'rxjs';
+
+@Component({
+  selector: 'cx-reorder-dialog',
+  templateUrl: './reorder-dialog.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ReorderDialogComponent implements OnInit, OnDestroy {
+  protected subscriptions = new Subscription();
+  iconTypes = ICON_TYPE;
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: true,
+    focusOnEscape: true,
+  };
+
+  cartModifications: CartModification[];
+
+  loading$ = new BehaviorSubject(false);
+
+  constructor(protected launchDialogService: LaunchDialogService) {}
+
+  ngOnInit() {
+    this.subscriptions.add(
+      this.launchDialogService.data$.subscribe((data: any) => {
```

Comment:
Could the subscription to data$ have been done in the template to simplify the template code?
What do you think?